Create simple UI kit module

ptesser opened the pull request:
Ho creato le basi per il modulo UI.

In pratica ho generato un modulo esterno al progetto principale usando la libreria UI esterna https://vmware.github.io/clarity.

Questo consente di rimpiazzare facilmente le componenti che poi andremmo ad usare perché noi di fatto usiamo il wrapper e non direttamente la libreria.

Ho solo implementato (e neanche credo completamente) l'elemento di input e combobox per vedere che l'integrazione funzionasse correttamente.

Propongo di mergiare questa cosa in maniera semplice e all'occorrenza estendere con nuovi elementi,

ptesser committed:
feat (scaffold): add empty dir for the front-end

ptesser committed:
chore (app): remove init view and change app name

ptesser committed:
feat (ui): add library as internal projects

Use `ng generate library ui --prefix=ui` to generate it.

Include also VMWare UI lib to use in our components.

ptesser committed:
feat (ui): introduce input

ptesser committed:
feat (ui): add basic form custom component

ptesser committed:
feat (ui): introduce select

ptesser committed:
feat (ui): introduce abstract class for ui form control

ptesser committed:
feat (ci): add test for travis tool with Angular build

ptesser committed:
fix (app) - remove linting errors

ptesser commented:
@LucaVannucci Nel mentre ho aggiunto anche Travis per provare le build e i linting su CI, però se preferisci sposto la cosa su un'altra branch.

LucaVannucci commented:
No va bene vai tranquillo

> Il giorno 7 dic 2018, alle ore 09:32, Paolo Tesser <notifications@github.com> ha scritto:
> 
> @LucaVannucci <https://github.com/LucaVannucci> Nel mentre ho aggiunto anche Travis per provare le build e i linting su CI, però se preferisci sposto la cosa su un'altra branch.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/Hopstok/rock-recepies/pull/17#issuecomment-445160013>, or mute the thread <https://github.com/notifications/unsubscribe-auth/Ao--xlNSvIjCEcoC2p4jxesMsssZSIBfks5u2iengaJpZM4ZHbo5>.
>

ptesser committed:
feat (swagger) : introduce templates and scripts to generate code from doc

ptesser committed:
feat (api) - Introduce doc of the APis

ptesser committed:
chore (tslint): introduce no-unused local to allow fix swagger codegen issues automatically

This rule is deprecated from 2.9 version of Typescript, but without it there isn't a way to automatically fix this behaviour.

ptesser committed:
feat (scripts): add TS lint fix scripts

ptesser committed:
chore (env): substitute remote with local

ptesser committed:
chore (models): remove old models

This is done to move them inside new /models directory.

ptesser committed:
chore (models): change directory from /app

ptesser committed:
chore (composer): improve autoloaders

Vannnu committed:
Merge pull request #21 from Hopstok/feature/new-structure

Clean up back-end codes

Commit:
feat(Implementation): add method that inmplement common interface

Commit:
feat(interface): add common interface

Commit:
feat(common): add pagination on getAll method

Commit:
feat(controller): add constant values for http return

Commit:
feat(Controller): add methods to handle the businnes logic

Commit:
feat(implementation): add method to implement the ingredient interface

Commit:
feat(interface): add ingredient interface

Commit:
feat(route): add crud endpoint for ingredient

Commit:
feat(Request): add ingredient request

Commit:
feat(Interface): change methods return type, added new method

Commit:
feat(Repository/Implementation): add new Method and change return type

Vannnu commented on front-end/projects/ui/src/lib/input/input.component.spec.ts:
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
```
Line

ptesser replied:
Vedi altro commento.

Vannnu commented on front-end/projects/ui/src/lib/select/select.component.spec.ts:
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
```
Line

ptesser replied:
Vedi altro commento più significativo.

Vannnu commented on front-end/projects/ui/src/lib/select/select.component.ts:
```diff
@@ -0,0 +1,18 @@
+import { Component, Input, OnInit } from '@angular/core';
+
```
Line

ptesser replied:
Idem

Vannnu commented on front-end/projects/ui/src/lib/ui.component.spec.ts:
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
```
Line

ptesser replied:
Idem

Vannnu commented on front-end/projects/ui/src/lib/ui.module.ts:
```diff
@@ -0,0 +1,36 @@
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { ClarityModule, ClrFormsModule } from '@clr/angular';
+
```
Line

ptesser replied:
La linea è usata per separare gli import esterni da quelli interni. Vale anche per le altre parti

Vannnu commented on front-end/projects/ui/src/lib/ui.module.ts:
```diff
@@ -0,0 +1,36 @@
+import { SelectComponent } from './select/select.component';
+import { UiComponent } from './ui.component';
+
+const COMPONENTS = [
```
Explain the behavior of the const

ptesser replied:
In che senso? è un array di componenti che uso sia nelle dichiarazioni del modulo sia negli export, ovvero quello che il modulo espone all'esterno.

ptesser commented on front-end/projects/ui/src/lib/input/input.component.ts:
```diff
@@ -0,0 +1,14 @@
+import { Component, OnInit } from '@angular/core';
+import { UiFormControl } from '../ui-form-control';
```
Qua devo separare, sistemo.

Commit:
fix(interface): fix methods type return

Commit:
fix(Implementation): fix param method types

Commit:
controller: add new services on ingredients entity

ptesser committed:
Merge pull request #11 from Hopstok/feature/front-end/angular-scaffold

Add Angular 7 scaffold and improve linting

Vannnu committed:
Merge pull request #24 from Hopstok/feature/commons

Implements common interface and implementation for repository pattern

Vannnu committed:
Merge pull request #25 from Hopstok/feature/ingredient

Implement Ingredient resource
